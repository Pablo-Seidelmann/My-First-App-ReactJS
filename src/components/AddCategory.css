.form-category
{
    margin: 50px;
    text-align: center;
}

.input-category
{
    color: grey;
    border-radius: 20px;
    border: 1px solid #38C98D;
    font-weight: bold;
    font-size: 15px;
    padding: 40px;
    text-align: center;
}

.btn-div
{
    position: relative;
    text-align: center;
    top: 20px; 
}

.btn-primary 
{
    border-radius: 5px;
    border: 1px solid grey;
    background-color: #38C98D;
    font-weight: bold;
    color: #fff;
    margin-left: 10px;
    padding: 8px 20px; 
}

.btn-primary:hover
{
    background-color:  #37b983;
    transition: 1.5s;
}

.btn-secondary
{
    border-radius: 5px;
    border: 1px solid grey;
    background-color: gray;
    font-weight: bold;
    color: #fff;
    margin-left: 10px;
    padding: 8px 20px; 
}