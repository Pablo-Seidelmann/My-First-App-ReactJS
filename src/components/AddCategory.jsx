import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import swal from 'sweetalert';

const AddCategory = ({ setCategories }) => {

    const Form = styled.form`
    text-align: center;
    margin: 50px;
`

    const Input = styled.input`
        padding: 40px;
        border-radius: 20px;
        border: 1px solid #38C98D;
        color: grey;
        font-weight: bold;
        text-align: center;
        font-size: 15px;
    `

    const InputButton = styled.button`
        margin-left: 10px;
        background-color: ${ props => props.primary ? '#38C98D' : 'grey' };
        color: #fff;
        padding: 8px 20px; 
        border-radius: 5px;
        border: 1px solid grey;
        font-weight: bold;

        &:hover
        {
            background-color: ${ props => props.hover ? '#808090' : null };
            transition: 1.5s;
        }
    `

    const DivButton = styled.div`
        text-align: center;
        position: relative;
        top: 20px;
    `

    const [ inputValue, setInputValue ] = useState('Escribe Aca!');

    const handleInputChange = ( e ) => setInputValue( e.target.value );

    const handleSubmit = ( e ) => {
        e.preventDefault();

        if( inputValue.length > 5 ) {
            setCategories( ( cats ) => [ ...cats, inputValue ] );
            setInputValue('');
        } else {
            swal({
                title: "A Ocurrido Algo Inesperado!",
                text: "Debe haber al menos 5 Caracteres!",
                icon: "warning",
                button: "Ok!",
              });
        };
    };
    
    return (

        <Form onSubmit = { handleSubmit } >
            
            <Input 
                type = 'text'
                value = { inputValue }
                onChange = { handleInputChange }
            />

            <DivButton>

            <InputButton primary type='submit' >Enviar</InputButton>

            <InputButton hover type='reset' >Cancelar</InputButton>

            </DivButton>

        </Form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
