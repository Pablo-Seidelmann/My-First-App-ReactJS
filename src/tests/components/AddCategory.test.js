import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import AddCategory from '../../components/AddCategory';

describe('Pruebas al Componente <AddCategory />', () => {

    const setCategories = jest.fn();

    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories = { setCategories } /> );
    })

    test('SnapShot al Componente', () => {
       
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('El valor del input debe ser un string', () => {

        const value = 'Hola Mundo';

        const input = wrapper.find( '#Input-Category' );

        input.simulate( 'change', { target: { value:value } } );

        expect( wrapper.find( 'p' ).text().trim() ).toBe( value );

        const valueInput = wrapper.find( '#Input-Category' ).prop( 'value' );

        expect( valueInput ).toBe( value );

    });

    test('El onChange debe estar de forma correctamente', () => {
        
        const value = 'Hola Mundo';

        const change = wrapper.find( '#Input-Category' );

        change.simulate('change', { target: { value:value } });

        expect( wrapper.find('p').text().trim() ).toBe( value );

    });
    
    test('El form debe de estar de manera correcta', () => {
        
        wrapper.find( '#Form-Category' ).simulate( 'submit', { preventDefault(){} } );

        expect( setCategories ).toHaveBeenCalled();

    });

    test('Los botones Submit deberian haberse llamado una vez', () => {
        
        const input = wrapper.find( '#Input-Category' );

        input.simulate( 'submit' );

        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('Deberia Enviar algo y luego el valor del input debe estar vacio', () => {
        
        const value = 'Hola Mundo';

        wrapper.find('#Input-Category').simulate('change', { target: { value: value } });

        wrapper.find('#Form-Category').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalledTimes( 1 )

        expect( wrapper.find( '#Input-Category' ).prop('value') ).toBe( ' '.trim() )

    });
    
    
    
    
    
});
