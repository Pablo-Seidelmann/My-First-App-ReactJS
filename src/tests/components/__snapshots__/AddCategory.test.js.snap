// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al Componente <AddCategory /> SnapShot al Componente 1`] = `
<styled.form
  id="Form-Category"
  onSubmit={[Function]}
>
  <styled.input
    id="Input-Category"
    onChange={[Function]}
    type="text"
    value="Escribe Aca!"
  />
  <p>
    Escribe Aca!
  </p>
   
  <styled.div>
    <styled.button
      id="Button-Category"
      primary={true}
      type="submit"
    >
      Enviar
    </styled.button>
    <styled.button
      hover={true}
      id="Button-Category"
      type="submit"
    >
      Cancelar
    </styled.button>
  </styled.div>
</styled.form>
`;
